nucypher/cli/characters/stake.py

```diff
@@ -34,63 +34,164 @@
-def _stake_options(func):
-    @_api_options
-    @click.option('--hw-wallet/--no-hw-wallet', default=False)  # TODO: Make True or deprecate.
-    @click.option('--beneficiary-address', help="Address of a pre-allocation beneficiary", type=EIP55_CHECKSUM_ADDRESS)
-    @click.option('--allocation-filepath', help="Path to individual allocation file", type=EXISTING_READABLE_FILE)
-    @functools.wraps(func)
-    def wrapper(*args, **kwargs):
-        return func(*args, **kwargs)
-
-    return wrapper
-
-
-# Args (poa, registry_filepath, config_file, provider_uri, staking_address, hw_wallet,
-#       beneficiary_address, allocation_filepath, worker_address)
-def _worker_options(func):
-    @_stake_options
-    @click.option('--worker-address', help="Address to assign as an Ursula-Worker", type=EIP55_CHECKSUM_ADDRESS)
-    @functools.wraps(func)
-    def wrapper(*args, **kwargs):
-        return func(*args, **kwargs)
-
-    return wrapper
+option_value = click.option('--value', help="Token value of stake", type=click.INT)
+option_lock_periods = click.option('--lock-periods', help="Duration of stake in periods.", type=click.INT)
+option_worker_address = click.option('--worker-address', help="Address to assign as an Ursula-Worker", type=EIP55_CHECKSUM_ADDRESS)
+
+
+class StakeHolderConfigOptions:
+
+    __option_name__ = 'config_options'
+
+    def __init__(self, provider_uri, poa, light, registry_filepath):
+        self.provider_uri = provider_uri
+        self.poa = poa
+        self.light = light
+        self.registry_filepath = registry_filepath
+
+    def create_config(self, emitter, config_file):
+        try:
+            return StakeHolderConfiguration.from_configuration_file(
+                emitter=emitter,
+                filepath=config_file,
+                provider_uri=self.provider_uri,
+                poa=self.poa,
+                light=self.light,
+                sync=False,
+                registry_filepath=self.registry_filepath)
+
+        except FileNotFoundError:
+            return actions.handle_missing_configuration_file(
+                character_config_class=StakeHolderConfiguration,
+                config_file=config_file)
+
+    def generate_config(self, config_root):
+
+        if self.provider_uri is None:
+            raise click.BadOptionUsage(
+                option_name="--provider-uri",
+                message="--provider-uri must be specified to create a new stakeholder")
+
+        return StakeHolderConfiguration.generate(
+            config_root=config_root,
+            provider_uri=self.provider_uri,
+            poa=self.poa,
+            light=self.light,
+            sync=False,
+            registry_filepath=self.registry_filepath)
+
+
+group_config_options = group_options(
+    StakeHolderConfigOptions,
+    provider_uri=option_provider_uri(),
+    poa=option_poa,
+    light=option_light,
+    registry_filepath=option_registry_filepath,
+    )
+
+
+class StakeHolderCharacterOptions:
```

Comment:
I think a renaming is in order here. `Stakeholder` is no longer a character, nor it's actually the relevant actor here (in other words, Stakeholder is something basically internal). 
I'd name this `StakerOptions`, simply, although since we have to differentiate here with the other one (currently named `PreallocationStakeHolderCharacterOptions`), and the main difference between them is the possibility of being an "authenticated" staker (i.e., being able to transact) or not (i.e., just to use `list` or `accounts`), maybe the naming should reflect this.
For instance, this group can be `StakerOptions`, while the other can be `TransactingStakerOptions`. 
@KPrasch what do you think?

Reply:
I agree with the reasoning, but I'll hold the renaming until we all agree on the final names.

Reply:
Renamed to `StakerOptions` and `TransactingStakerOptions`.